tensorflow_probability/python/stats/sample_stats.py

```diff
@@ -645,6 +807,104 @@ def variance(x, sample_axis=0, keepdims=False, name=None):
         x, y=None, sample_axis=sample_axis, event_axis=None, keepdims=keepdims)
 
 
+VarianceState = collections.namedtuple('VarianceState',
+                                       'num_samples, mean, m2, dtype')
```

Comment:
Is there a better name than `m2`? I have used "raw_variance_estimate"?

Reply:
Perhaps `sum_squared_residuals` given that `m2` isn't exactly an estimate of the variance yet? I'll refactor to that for now but open to naming suggestions (:

Reply:
Sounds good!